### ISSUE TITLE
MAVROS regression connecting to 3DR Solo drones over WIFI

### ISSUE THREAD

**Issue by sondreengebraten** [mavlink/mavros#1017]
### Issue details

Unable to connect to autopilot on 3DR Solo drone using UDP over controller access point link. This used to work perfectly, but upon updating MAVROS to 0.23.3 this breaks. I need to connect to the autopilot to set ardupilot parameters. Currently heartbeats are received from the autopilot, but no information appears to be sent back to the autopilot from the host computer. This causes parameter list to never be loaded, home position is never set.  Note that this is clearly a regression issue or perhaps something to do with a new way of configuring MAVROS, as 15+ 3DR Solos have already been successfully configured this way using an older MAVROS version. MAVROS was installed using APT. Any suggestions would be most welcome.

### MAVROS version and platform

Mavros: 0.23.3
ROS: Kinetic
Linux Mint 18.3 / Ubuntu: 16.04

### Autopilot type and version

[ x] ArduPilot

Version: ArdupilotSolo-2.3.0-1

### Node logs

_roslaunch mavros apm.launch fcu_url:=udp://0.0.0.0@10.1.1.10:14560 gcu_url:=udp://@localhost_

```started roslaunch server http://mp005:35245/

SUMMARY
========

CLEAR PARAMETERS
 * /mavros/

PARAMETERS
 * /mavros/cmd/use_comp_id_system_control: False
 * /mavros/conn/heartbeat_rate: 1.0
 * /mavros/conn/system_time_rate: 0.0
 * /mavros/conn/timeout: 10.0
 * /mavros/conn/timesync_rate: 0.0
 * /mavros/distance_sensor/rangefinder_pub/field_of_view: 0.0
 * /mavros/distance_sensor/rangefinder_pub/frame_id: lidar
 * /mavros/distance_sensor/rangefinder_pub/id: 0
 * /mavros/distance_sensor/rangefinder_pub/send_tf: False
 * /mavros/distance_sensor/rangefinder_pub/sensor_position/x: 0.0
 * /mavros/distance_sensor/rangefinder_pub/sensor_position/y: 0.0
 * /mavros/distance_sensor/rangefinder_pub/sensor_position/z: -0.1
 * /mavros/distance_sensor/rangefinder_sub/id: 1
 * /mavros/distance_sensor/rangefinder_sub/orientation: PITCH_270
 * /mavros/distance_sensor/rangefinder_sub/subscriber: True
 * /mavros/fake_gps/eph: 2.0
 * /mavros/fake_gps/epv: 2.0
 * /mavros/fake_gps/fix_type: 3
 * /mavros/fake_gps/geo_origin/alt: 408.0
 * /mavros/fake_gps/geo_origin/lat: 47.3667
 * /mavros/fake_gps/geo_origin/lon: 8.55
 * /mavros/fake_gps/gps_rate: 5.0
 * /mavros/fake_gps/mocap_transform: True
 * /mavros/fake_gps/satellites_visible: 5
 * /mavros/fake_gps/tf/child_frame_id: fix
 * /mavros/fake_gps/tf/frame_id: map
 * /mavros/fake_gps/tf/listen: False
 * /mavros/fake_gps/tf/rate_limit: 10.0
 * /mavros/fake_gps/tf/send: False
 * /mavros/fake_gps/use_mocap: True
 * /mavros/fake_gps/use_vision: False
 * /mavros/fcu_protocol: v2.0
 * /mavros/fcu_url: udp://0.0.0.0@10....
 * /mavros/gcs_url: 
 * /mavros/global_position/child_frame_id: base_link
 * /mavros/global_position/frame_id: map
 * /mavros/global_position/rot_covariance: 99999.0
 * /mavros/global_position/tf/child_frame_id: base_link
 * /mavros/global_position/tf/frame_id: map
 * /mavros/global_position/tf/global_frame_id: earth
 * /mavros/global_position/tf/send: False
 * /mavros/global_position/use_relative_alt: True
 * /mavros/image/frame_id: px4flow
 * /mavros/imu/angular_velocity_stdev: 0.000349065850399
 * /mavros/imu/frame_id: base_link
 * /mavros/imu/linear_acceleration_stdev: 0.0003
 * /mavros/imu/magnetic_stdev: 0.0
 * /mavros/imu/orientation_stdev: 1.0
 * /mavros/local_position/frame_id: map
 * /mavros/local_position/tf/child_frame_id: base_link
 * /mavros/local_position/tf/frame_id: map
 * /mavros/local_position/tf/send: False
 * /mavros/local_position/tf/send_fcu: False
 * /mavros/mission/pull_after_gcs: True
 * /mavros/mocap/use_pose: True
 * /mavros/mocap/use_tf: False
 * /mavros/odometry/estimator_type: 3
 * /mavros/odometry/frame_tf/desired_frame: ned
 * /mavros/plugin_blacklist: ['actuator_contro...
 * /mavros/plugin_whitelist: []
 * /mavros/px4flow/frame_id: px4flow
 * /mavros/px4flow/ranger_fov: 0.118682389136
 * /mavros/px4flow/ranger_max_range: 5.0
 * /mavros/px4flow/ranger_min_range: 0.3
 * /mavros/safety_area/p1/x: 1.0
 * /mavros/safety_area/p1/y: 1.0
 * /mavros/safety_area/p1/z: 1.0
 * /mavros/safety_area/p2/x: -1.0
 * /mavros/safety_area/p2/y: -1.0
 * /mavros/safety_area/p2/z: -1.0
 * /mavros/setpoint_accel/send_force: False
 * /mavros/setpoint_attitude/reverse_thrust: False
 * /mavros/setpoint_attitude/tf/child_frame_id: target_attitude
 * /mavros/setpoint_attitude/tf/frame_id: map
 * /mavros/setpoint_attitude/tf/listen: False
 * /mavros/setpoint_attitude/tf/rate_limit: 50.0
 * /mavros/setpoint_attitude/use_quaternion: False
 * /mavros/setpoint_position/mav_frame: LOCAL_NED
 * /mavros/setpoint_position/tf/child_frame_id: target_position
 * /mavros/setpoint_position/tf/frame_id: map
 * /mavros/setpoint_position/tf/listen: False
 * /mavros/setpoint_position/tf/rate_limit: 50.0
 * /mavros/setpoint_velocity/mav_frame: LOCAL_NED
 * /mavros/startup_px4_usb_quirk: False
 * /mavros/sys/disable_diag: False
 * /mavros/sys/min_voltage: 10.0
 * /mavros/target_component_id: 1
 * /mavros/target_system_id: 1
 * /mavros/tdr_radio/low_rssi: 40
 * /mavros/time/time_ref_source: fcu
 * /mavros/time/timesync_avg_alpha: 0.6
 * /mavros/time/timesync_mode: NONE
 * /mavros/vibration/frame_id: base_link
 * /mavros/vision_pose/tf/child_frame_id: vision_estimate
 * /mavros/vision_pose/tf/frame_id: map
 * /mavros/vision_pose/tf/listen: False
 * /mavros/vision_pose/tf/rate_limit: 10.0
 * /mavros/vision_speed/listen_twist: False
 * /rosdistro: kinetic
 * /rosversion: 1.12.13

NODES
  /
    mavros (mavros/mavros_node)

auto-starting new master
process[master]: started with pid [10578]
ROS_MASTER_URI=http://localhost:11311

setting /run_id to e1194f68-473a-11e8-8fb9-801934c7aa06
process[rosout-1]: started with pid [10591]
started core service [/rosout]
process[mavros-2]: started with pid [10609]
[ INFO] [1524517885.040035037]: FCU URL: udp://0.0.0.0@10.1.1.10:14560
[ INFO] [1524517885.041504681]: udp0: Bind address: 0.0.0.0:14555
[ INFO] [1524517885.041561124]: udp0: Remote address: 10.1.1.10:14560
[ INFO] [1524517885.041678774]: GCS bridge disabled
[ INFO] [1524517885.051612786]: Plugin 3dr_radio loaded
[ INFO] [1524517885.053583537]: Plugin 3dr_radio initialized
[ INFO] [1524517885.053618691]: Plugin actuator_control blacklisted
[ INFO] [1524517885.053648283]: Plugin altitude blacklisted
[ INFO] [1524517885.053715984]: Plugin command loaded
[ INFO] [1524517885.058848255]: Plugin command initialized
[ INFO] [1524517885.058881447]: Plugin ftp blacklisted
[ INFO] [1524517885.058966547]: Plugin global_position loaded
[ INFO] [1524517885.086398049]: Plugin global_position initialized
[ INFO] [1524517885.086460757]: Plugin hil blacklisted
[ INFO] [1524517885.086600788]: Plugin home_position loaded
[ INFO] [1524517885.092021494]: Plugin home_position initialized
[ INFO] [1524517885.092135860]: Plugin imu loaded
[ INFO] [1524517885.101450918]: Plugin imu initialized
[ INFO] [1524517885.101544928]: Plugin local_position loaded
[ INFO] [1524517885.107423770]: Plugin local_position initialized
[ INFO] [1524517885.107502500]: Plugin manual_control loaded
[ INFO] [1524517885.111960742]: Plugin manual_control initialized
[ INFO] [1524517885.112068422]: Plugin param loaded
[ INFO] [1524517885.114985462]: Plugin param initialized
[ INFO] [1524517885.115079575]: Plugin rc_io loaded
[ INFO] [1524517885.120747306]: Plugin rc_io initialized
[ INFO] [1524517885.120792236]: Plugin safety_area blacklisted
[ INFO] [1524517885.120867721]: Plugin setpoint_accel loaded
[ INFO] [1524517885.124842739]: Plugin setpoint_accel initialized
[ INFO] [1524517885.124928154]: Plugin setpoint_attitude loaded
[ INFO] [1524517885.138010201]: Plugin setpoint_attitude initialized
[ INFO] [1524517885.138149924]: Plugin setpoint_position loaded
[ INFO] [1524517885.160145572]: Plugin setpoint_position initialized
[ INFO] [1524517885.160299628]: Plugin setpoint_raw loaded
[ INFO] [1524517885.175727893]: Plugin setpoint_raw initialized
[ INFO] [1524517885.175860667]: Plugin setpoint_velocity loaded
[ INFO] [1524517885.185939614]: Plugin setpoint_velocity initialized
[ INFO] [1524517885.186124372]: Plugin sys_status loaded
[ INFO] [1524517885.195071359]: Plugin sys_status initialized
[ INFO] [1524517885.195217507]: Plugin sys_time loaded
[ INFO] [1524517885.199706476]: TM: Timesync mode: NONE
[ INFO] [1524517885.201090845]: Plugin sys_time initialized
[ INFO] [1524517885.201236691]: Plugin vfr_hud loaded
[ INFO] [1524517885.202885628]: Plugin vfr_hud initialized
[ INFO] [1524517885.202988973]: Plugin waypoint loaded
[ INFO] [1524517885.208901928]: Plugin waypoint initialized
[ INFO] [1524517885.208948112]: Built-in SIMD instructions: SSE, SSE2
[ INFO] [1524517885.208987673]: Built-in MAVLink package version: 2018.3.7
[ INFO] [1524517885.209033292]: Known MAVLink dialects: common ardupilotmega ASLUAV autoquad icarous matrixpilot paparazzi slugs standard uAvionix ualberta
[ INFO] [1524517885.209167097]: MAVROS started. MY ID 1.240, TARGET ID 1.1
[ INFO] [1524517886.204166884]: udp0: Remote address: 10.1.1.10:14560
[ INFO] [1524517886.304663202]: IMU: Raw IMU message used.
[ WARN] [1524517886.304855554]: IMU: linear acceleration on RAW_IMU known on APM only.
[ WARN] [1524517886.304938086]: IMU: ~imu/data_raw stores unscaled raw acceleration report.
[ INFO] [1524517886.568014323]: CON: Got HEARTBEAT, connected. FCU: ArduPilotMega / ArduCopter
[ WARN] [1524517886.770828206]: RADIO_STATUS not from 3DR modem?
[ERROR] [1524517887.576848796]: FCU: PreArm: Need 3D Fix
[ WARN] [1524517888.570652531]: VER: broadcast request timeout, retries left 4
[ WARN] [1524517889.573248390]: VER: broadcast request timeout, retries left 3
[ERROR] [1524517889.608935518]: FCU: PreArm: Need 3D Fix
[ERROR] [1524517891.642456556]: FCU: PreArm: Need 3D Fix
[ERROR] [1524517893.671931876]: FCU: PreArm: Need 3D Fix
[ WARN] [1524517895.572058030]: VER: unicast request timeout, retries left 2
[ERROR] [1524517895.704443934]: FCU: PreArm: Need 3D Fix
[ INFO] [1524517896.569276295]: HP: requesting home position
[ERROR] [1524517897.733933942]: FCU: PreArm: Need 3D Fix
[ERROR] [1524517899.801182698]: FCU: PreArm: Need 3D Fix
[ WARN] [1524517900.576460800]: VER: unicast request timeout, retries left 1
[ WARN] [1524517900.576764512]: PR: request list timeout, retries left 2
[ WARN] [1524517901.577397553]: PR: request list timeout, retries left 1
[ERROR] [1524517901.797810149]: FCU: PreArm: Need 3D Fix
[ WARN] [1524517902.573937972]: WP: timeout, retries left 2
[ WARN] [1524517902.577858583]: PR: request list timeout, retries left 0
[ WARN] [1524517903.574287066]: WP: timeout, retries left 1
[ERROR] [1524517903.832529146]: FCU: PreArm: Need 3D Fix
[ WARN] [1524517904.574711732]: WP: timeout, retries left 0
[ERROR] [1524517905.575718110]: WP: timed out.
[ WARN] [1524517905.580999137]: VER: unicast request timeout, retries left 0
[ERROR] [1524517905.864160048]: FCU: PreArm: Need 3D Fix
[ INFO] [1524517906.569368018]: HP: requesting home position
[ERROR] [1524517907.893665502]: FCU: PreArm: Need 3D Fix
[ERROR] [1524517909.925731782]: FCU: PreArm: Need 3D Fix
[ WARN] [1524517910.585127909]: VER: your FCU don't support AUTOPILOT_VERSION, switched to default capabilities
[ERROR] [1524517911.957250697]: FCU: PreArm: Need 3D Fix
[ERROR] [1524517913.992109825]: FCU: PreArm: Need 3D Fix
[ERROR] [1524517916.023531065]: FCU: PreArm: Need 3D Fix
[ INFO] [1524517916.568837552]: HP: requesting home position
[ WARN] [1524517917.770396122]: RADIO_STATUS not from 3DR modem?
[ERROR] [1524517918.054584646]: FCU: PreArm: Need 3D Fix
[ERROR] [1524517920.083688173]: FCU: PreArm: Need 3D Fix
[ERROR] [1524517922.122587226]: FCU: PreArm: Need 3D Fix
[ERROR] [1524517924.155908631]: FCU: PreArm: Need 3D Fix
```

### Diagnostics

```header: 
  seq: 50
  stamp: 
    secs: 1524517949
    nsecs: 586038799
  frame_id: ''
status: 
  - 
    level: 0
    name: "mavros: FCU connection"
    message: "connected"
    hardware_id: "udp://0.0.0.0@10.1.1.10:14560"
    values: 
      - 
        key: "Received packets:"
        value: "5352"
      - 
        key: "Dropped packets:"
        value: "0"
      - 
        key: "Buffer overruns:"
        value: "0"
      - 
        key: "Parse errors:"
        value: "0"
      - 
        key: "Rx sequence number:"
        value: "163"
      - 
        key: "Tx sequence number:"
        value: "79"
      - 
        key: "Rx total bytes:"
        value: "158485"
      - 
        key: "Tx total bytes:"
        value: "1896"
      - 
        key: "Rx speed:"
        value: "3709.000000"
      - 
        key: "Tx speed:"
        value: "21.000000"
  - 
    level: 0
    name: "mavros: GPS"
    message: "3D fix"
    hardware_id: "udp://0.0.0.0@10.1.1.10:14560"
    values: 
      - 
        key: "Satellites visible"
        value: "12"
      - 
        key: "Fix type"
        value: "3"
      - 
        key: "EPH (m)"
        value: "1.83"
      - 
        key: "EPV (m)"
        value: "Unknown"
  - 
    level: 0
    name: "mavros: Heartbeat"
    message: "Normal"
    hardware_id: "udp://0.0.0.0@10.1.1.10:14560"
    values: 
      - 
        key: "Heartbeats since startup"
        value: "63"
      - 
        key: "Frequency (Hz)"
        value: "1.037097"
      - 
        key: "Vehicle type"
        value: "Quadrotor"
      - 
        key: "Autopilot type"
        value: "ArduPilotMega / ArduCopter"
      - 
        key: "Mode"
        value: "LOITER"
      - 
        key: "System status"
        value: "Standby"
  - 
    level: 0
    name: "mavros: System"
    message: "Normal"
    hardware_id: "udp://0.0.0.0@10.1.1.10:14560"
    values: 
      - 
        key: "Sensor present"
        value: "0x0061FC2F"
      - 
        key: "Sensor enabled"
        value: "0x0061FC2F"
      - 
        key: "Sensor helth"
        value: "0x0061FC2F"
      - 
        key: "3D gyro"
        value: "Ok"
      - 
        key: "3D accelerometer"
        value: "Ok"
      - 
        key: "3D magnetometer"
        value: "Ok"
      - 
        key: "absolute pressure"
        value: "Ok"
      - 
        key: "GPS"
        value: "Ok"
      - 
        key: "3D angular rate control"
        value: "Ok"
      - 
        key: "attitude stabilization"
        value: "Ok"
      - 
        key: "yaw position"
        value: "Ok"
      - 
        key: "z/altitude control"
        value: "Ok"
      - 
        key: "x/y position control"
        value: "Ok"
      - 
        key: "motor outputs / control"
        value: "Ok"
      - 
        key: "rc receiver"
        value: "Ok"
      - 
        key: "AHRS subsystem health"
        value: "Ok"
      - 
        key: "Terrain subsystem health"
        value: "Ok"
      - 
        key: "CPU Load (%)"
        value: "24.4"
      - 
        key: "Drop rate (%)"
        value: "0.0"
      - 
        key: "Errors comm"
        value: "0"
      - 
        key: "Errors count #1"
        value: "0"
      - 
        key: "Errors count #2"
        value: "0"
      - 
        key: "Errors count #3"
        value: "0"
      - 
        key: "Errors count #4"
        value: "0"
  - 
    level: 0
    name: "mavros: Battery"
    message: "Normal"
    hardware_id: "udp://0.0.0.0@10.1.1.10:14560"
    values: 
      - 
        key: "Voltage"
        value: "15.11"
      - 
        key: "Current"
        value: "0.6"
      - 
        key: "Remaining"
        value: "30.0"
  - 
    level: 1
    name: "mavros: 3DR Radio"
    message: "Low RSSI"
    hardware_id: "udp://0.0.0.0@10.1.1.10:14560"
    values: 
      - 
        key: "RSSI"
        value: "0"
      - 
        key: "RSSI (dBm)"
        value: "-127.0"
      - 
        key: "Remote RSSI"
        value: "228"
      - 
        key: "Remote RSSI (dBm)"
        value: "-7.0"
      - 
        key: "Tx buffer (%)"
        value: "0"
      - 
        key: "Noice level"
        value: "0"
      - 
        key: "Remote noice level"
        value: "0"
      - 
        key: "Rx errors"
        value: "4"
      - 
        key: "Fixed"
        value: "0"
---
```

### Check ID

```
OK. I got messages from 1:1.

---
Received 1273 messages, from 2 addresses
sys:comp   list of messages
 10:0     109
  1:1     0, 1, 2, 129, 136, 137, 152, 150, 24, 27, 29, 30, 33, 35, 36, 165, 42, 178, 62, 193, 74, 163, 253, 116, 252, 125
```

**Comment by khancyr** [comment]
hello, change protocol to mavlink v1 and it should works

**Comment by sondreengebraten** [comment]
How do I do that?

**Comment by sondreengebraten** [comment]
If this is a problem others might run into too, I suggest including the following pull request #1018

**Comment by sondreengebraten** [comment]
@khancyr  Worked like a charm :) Thank you!

**Comment by TSC21** [comment]
Fixed in #1018.
